### PR TITLE
Update OTEL pdata dependency

### PR DESCRIPTION
**Pull request by Aneurysm9**
Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>

### CODE REVIEW

**Review by jacobmarble — CHANGES_REQUESTED** [review]
Hi, and thanks for this. Can you explain why this updates OpenTelemetry to `v0.38.1-0.20211103215828-cffbecb2ac9e` rather than `v0.38.0`?

Please adjust the commit message and PR title to conform to semantic format, as indicated by CI.

Please sign the InfluxData CLA https://www.influxdata.com/legal/cla/
(this is not stated in the README, so I should add that to this and a couple related repositories)

**Comment by Aneurysm9** [comment]
It updates to the commit where the change was introduced and to which the contrib repo is going to be updated.  This repo sits in the middle and blocks the update to contrib until it is updated.  I don't think we can wait until the next core release to update contrib, nor can we release core every time the parts of it that this repo depends on change.

I'm not sure whether I am able to sign the influx CLA and don't have time at the moment to chase approvals.  Can you please effect this change ASAP through commits under your signature.

**Pull request closed without merging**
